src/test/shared/utilities/workspaceUtils.test.ts

```diff
@@ -0,0 +1,82 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import * as path from 'path'
+import * as sinon from 'sinon'
+import * as vscode from 'vscode'
+import { findParentProjectFile } from '../../../shared/utilities/workspaceUtils'
+
+describe('findParentProjectFile', async () => {
+    const workspaceDir = path.join('code', 'workspace')
+    before(() => {
+        const sandbox = sinon.createSandbox()
+        sandbox.stub(vscode.workspace, 'getWorkspaceFolder').returns({
+            name: 'tempFolder',
+            index: 0,
+            uri: vscode.Uri.parse(workspaceDir),
+        })
```

Comment:
if `findParentProjectFile()` took a `vscode.Workspace` parameter then this mock stuff wouldn't be needed. That also makes `findParentProjectFile()` more flexible.

Reply:
Moving these to integration tests to avoid mocks and get a higher fidelity picture of what these tests will look like.

Reply:
Remember that ideally you'd want both unit tests and integration tests. It's expected that there's some overlap.

I'm thinking that creating all these variations of project structures on the real fs would be more tedious than mocking. IMO it would be good to do all the variations in mock and a single integration to test the wiring.